libnavigation-ui/src/main/java/com/mapbox/navigation/ui/camera/NavigationCamera.java

```diff
@@ -218,8 +220,17 @@ public void resetCameraPositionWith(@TrackingMode int trackingMode) {
    */
   public void showRouteOverview(@NonNull int[] padding) {
     updateCameraTrackingMode(NAVIGATION_TRACKING_MODE_NONE);
+    DirectionsRoute currentRoute;
+    if (currentRouteProgress != null) {
+      currentRoute = currentRouteProgress.getRoute();
+    } else if (currentRouteInformation != null) {
+      currentRoute = currentRouteInformation.getRoute();
+    } else {
+      Timber.e("Unable to show route overview, the route is null.");
+      return;
```

Comment:
I just checked the flow in `NavigationView`, with this fix, we will see the `Recenter` icon pops up and the route is still in tracking mode.

since there will be null case as you already saw in #3593, we might want to return a boolean `showRouteOverview` result to the caller and returns `false` here. Then `NavigationView` will only show recenter icon and hide the wayName view when `showRouteOverview` succeeds.
Like in `NavigationPresenter`:
```java
  void onRouteOverviewClick() {
    if (view.updateCameraRouteOverview()) {
        view.setWayNameActive(false);
        view.setWayNameVisibility(false);
        view.showRecenterBtn();
    }
  }
```

Reply:
Nice catch! Unfortunately, this would be a semver-major change. I opted to deprecate the method instead.